### PR TITLE
Small fix for colorwallmounted  and colorfacedir nodes

### PR DESCRIPTION
**Pull request by juhdanad**
Contains:
- correct node placement prediction for attached wallmounted colored nodes,
- correct placement direction for colorfacedir and colorwallmounted nodes and
- correct detatch mechanism for attached colored wallmounted nodes

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -884,7 +884,8 @@ bool nodePlacementPrediction(Client &client,
 			};
 			v3s16 pp;
 
-			if (nodedef->get(id).param_type_2 == CPT2_WALLMOUNTED)
+			if (nodedef->get(id).param_type_2 == CPT2_WALLMOUNTED ||
+					nodedef->get(id).param_type_2 == CPT2_COLORED_WALLMOUNTED)
 				pp = p + wallmounted_dirs[param2];
```
Comment (paramat) [inline src/game.cpp]: Do you need to take the first 3 bits of param2 here?

File: builtin/game/falling.lua [hunk 2]
```diff
@@ -134,7 +134,8 @@ end
 function builtin_shared.check_attached_node(p, n)
 	local def = core.registered_nodes[n.name]
 	local d = {x = 0, y = 0, z = 0}
-	if def.paramtype2 == "wallmounted" then
+	if def.paramtype2 == "wallmounted" or
+			def.paramtype2 == "colorwallmounted" then
 		-- The fallback vector here is in case 'wallmounted to dir' is nil due
 		-- to voxelmanip placing a wallmounted node without resetting a
 		-- pre-existing param2 value that is out-of-range for wallmounted.
```
Comment (paramat) [inline builtin/game/falling.lua]: Line 143 below: need to take the first 3 bits of param2?

**Comment by paramat** [comment]
Otherwise looks fine.

File: builtin/game/falling.lua (hunk 2 above)
Comment (juhdanad) [inline builtin/game/falling.lua]: No, `wallmounted_to_dir` does that.

**Comment by paramat** [comment]
:+1:

**Comment by paramat** [comment]
Sorry, accidental close.

**Comment by sofar** [comment]
fine with me :+1:

**Comment by paramat** [comment]
00123ee04d19ecc98e0a6a9255e5697a78167360

**Pull request closed without merging**